HotSpot/View Controllers/BookingViewController.m

```diff
@@ -83,4 +100,196 @@ - (IBAction)confirmClicked:(id)sender {
         }];
 }
 
+- (nonnull __kindof UICollectionViewCell *)collectionView:(nonnull UICollectionView *)collectionView
+                                   cellForItemAtIndexPath:(nonnull NSIndexPath *)indexPath {
+    TimeCell *cell = [collectionView dequeueReusableCellWithReuseIdentifier:@"TimeCell"
+                                                               forIndexPath:indexPath];
+    TimeSlot *timeSlot = self.timeSlots[indexPath.item];
+    [cell setTime:timeSlot];
+    if (!timeSlot.available) {
+        cell.backgroundColor = [UIColor colorWithRed:1.0 green:.2 blue:.4 alpha:1.0];
+    }
+    else if (timeSlot.chosen) {
+        cell.backgroundColor = [UIColor colorWithRed:0 green:.4 blue:1.0 alpha:1.0];
+    }
+    else {
+        cell.backgroundColor = [UIColor colorWithRed:0 green:.8 blue:.2 alpha:.6];
+    }
+    return cell;
+}
+
+- (NSInteger)collectionView:(nonnull UICollectionView *)collectionView numberOfItemsInSection:(NSInteger)section { 
+    return 24 * 4;
```

Comment:
you can directly return 96 here and leave a comment about how you got to that number